### ISSUE TITLE
Client Design Discussion

### ISSUE THREAD

**Issue by natalieschwarz2020** [hackwitus/pob-hdb-hackathon#2]
Lets start abstracting the Client in this thread. Lets go off this point: 
`The core functionality is tracking who takes what so on client side we need to be able to record the person's info, the item(s), the time in and out. We want to know what's available and what isn't. Maybe know who owns what like in our case we had to know if a item belonged to MLH, IEEE, or Tech sandbox`
So based on this statement, we need as a bare minimum:
- a login/sign up page, taking in the student's name and ID
- a page for selecting/borrowing items
    - potentially a dialog for the actual sign-out form

Let my know if anything should be added, after all, this is just a bare skeleton. Without a doubt, this should be a single-page application, and so a framework discussion should start here as well. I'm strongest with Angular2, so it has my vote but I'm willing to learn otherwise.

**Comment by Ethan-Arrowood** [comment]
I say we do not do a login/sign up page at first. Auth systems are not easy to implement and I’d rather focus on the utility of POB application. If we have time at the end (which I doubt) we can consider implementing an auth Page. For now I think we just have ‘users’ enter in their personal details when they want to borrow something

**Comment by natalieschwarz2020** [comment]
@Ethan-Arrowood That actually makes a ton of sense, just to make that part of the form instead of a separate entity

**Comment by Ethan-Arrowood** [comment]
Exactly. So what I’m imagining is a pretty simple SPA that has a menu. That menu contains links to different ‘pages’ (but are actually just different components. You know what I mean). Each component or view would be a different function for the app. 

1. Manage inventory of items. (Add, remove, edit)
2. Borrow an item (this would probably include a sorta menu to browse inventory and then a form to ‘check out’ an item/items)
3. Return an item (would maybe bring up list of borrowed things, select what’s being returned and then everything updates itself)

**Comment by natalieschwarz2020** [comment]
I agree with the idea of an inventory view, I think it would be smart to have it also function as a returning option(just by selecting/swiping the item, similar to deleting an email). Both the inventory and the list of items to be borrowed could be the same component with different flavors.

**Comment by Ethan-Arrowood** [comment]
Interesting idea, if that’s something you think you can implement I say go for it 🎊 Id encourage you to consult with everyone else before moving forward to get varrying opinions.

**Comment by ShawnToubeau** [comment]
![main-page](https://user-images.githubusercontent.com/22332636/35783118-631d01c4-09d0-11e8-8074-f7d65e4c4576.jpg)
![welcome-page optional](https://user-images.githubusercontent.com/22332636/35783119-632ae5be-09d0-11e8-88d7-d4d273eab9f6.jpg)
I came up with these two concepts, let me know what you think!

**Comment by ShawnToubeau** [comment]
![item-info](https://user-images.githubusercontent.com/22332636/35783275-7e1cf2fc-09d2-11e8-9455-5fb7d77badf7.jpg)
This too

**Comment by natalieschwarz2020** [comment]
I'm a fan of the Admin user idea, that will be super useful for keeping track of who specifically has taken out what. 

How will the users actually check out items? I'm thinking if we have the Welcome page it will be as simple as just clicking on the item with a small confirmation dialog. Thoughts, everyone?

**Comment by ShawnToubeau** [comment]
We could have like a virtual "shopping cart" where people can click on individual items and add them to their cart

**Comment by natalieschwarz2020** [comment]
I love that idea, that way it can be super simple to add items to the cart, and we could just have a "checkout" component

**Comment by ShawnToubeau** [comment]
Like in the last image, imagine a button that says "check out item" under the details section

**Comment by ShawnToubeau** [comment]
Or right next to the item name

**Comment by natalieschwarz2020** [comment]
It could be a button in the corner of the item list, maybe with a badge saying how many items you selected, so you could select more than one item and check them out all at once. That being said, I  still like the idea of the item's transaction history being public

**Comment by ShawnToubeau** [comment]
Exactly! I think this gives us a good idea of what we need so we should get started building it 🙂
